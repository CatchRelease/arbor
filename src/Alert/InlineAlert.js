import AlertContent from './AlertContent';

const InlineAlert = (props) => <AlertContent oneLine p="0" {...props} />;

export default InlineAlert;
