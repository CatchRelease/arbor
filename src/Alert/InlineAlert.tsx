import { FC } from 'react';
import AlertContent, { Props as AlertContentProps } from './AlertContent';

type Props = AlertContentProps;

const InlineAlert: FC<Props> = ({ oneLine = true, ...props }) => (
  <AlertContent p="0" {...{ oneLine, ...props }} />
);

export default InlineAlert;
