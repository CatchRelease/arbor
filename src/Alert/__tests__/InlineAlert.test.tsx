import InlineAlert from '../InlineAlert';
import behavesLikeAnAlert from './behavesLikeAnAlert';

describe('<InlineAlert />', () => {
  behavesLikeAnAlert(InlineAlert);
});
