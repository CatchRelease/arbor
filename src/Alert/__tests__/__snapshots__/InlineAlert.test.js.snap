// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineAlert /> property renders a banner alert with danger intent 1`] = `
.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-6 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr;
}

.emotion-0 {
  color: #D93838;
  font-size: 0.875rem;
}

<div
  className="emotion-6 emotion-7"
  display="grid"
>
  <div
    aria-label="danger"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-error emotion-0 emotion-1"
      color="intent.danger.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-4 emotion-5"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
</div>
`;

exports[`<InlineAlert /> property renders a banner alert with info intent 1`] = `
.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-6 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr;
}

.emotion-0 {
  color: #4263EB;
  font-size: 0.875rem;
}

<div
  className="emotion-6 emotion-7"
  display="grid"
>
  <div
    aria-label="info"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-info-sign emotion-0 emotion-1"
      color="intent.info.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-4 emotion-5"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
</div>
`;

exports[`<InlineAlert /> property renders a banner alert with no intent 1`] = `
.emotion-2 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
}

.emotion-0 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

<div
  className="emotion-2 emotion-3"
  display="grid"
>
  <div
    className="emotion-0 emotion-1"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
</div>
`;

exports[`<InlineAlert /> property renders a banner alert with success intent 1`] = `
.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-6 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr;
}

.emotion-0 {
  color: #29863A;
  font-size: 0.875rem;
}

<div
  className="emotion-6 emotion-7"
  display="grid"
>
  <div
    aria-label="success"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-tick-circle emotion-0 emotion-1"
      color="intent.success.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-4 emotion-5"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
</div>
`;

exports[`<InlineAlert /> property renders a banner alert with warning intent 1`] = `
.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-6 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr;
}

.emotion-0 {
  color: #C55205;
  font-size: 0.875rem;
}

<div
  className="emotion-6 emotion-7"
  display="grid"
>
  <div
    aria-label="warning"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-warning-sign emotion-0 emotion-1"
      color="intent.warning.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-4 emotion-5"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
</div>
`;

exports[`<InlineAlert /> with details property renders a banner alert, with danger intent, with details 1`] = `
.emotion-6 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #5B5D61;
  font-size: 0.875rem;
  line-height: 1.5;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-8 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr;
}

.emotion-0 {
  color: #D93838;
  font-size: 0.875rem;
}

<div
  className="emotion-8 emotion-9"
  display="grid"
>
  <div
    aria-label="danger"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-error emotion-0 emotion-1"
      color="intent.danger.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-6 emotion-7"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
    <span
      className="emotion-4 emotion-5"
      color="monochrome.grey80"
      fontSize="size4"
    >
       
      Here are some alert details
    </span>
  </div>
</div>
`;

exports[`<InlineAlert /> with details property renders a banner alert, with info intent, with details 1`] = `
.emotion-6 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #5B5D61;
  font-size: 0.875rem;
  line-height: 1.5;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-8 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr;
}

.emotion-0 {
  color: #4263EB;
  font-size: 0.875rem;
}

<div
  className="emotion-8 emotion-9"
  display="grid"
>
  <div
    aria-label="info"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-info-sign emotion-0 emotion-1"
      color="intent.info.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-6 emotion-7"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
    <span
      className="emotion-4 emotion-5"
      color="monochrome.grey80"
      fontSize="size4"
    >
       
      Here are some alert details
    </span>
  </div>
</div>
`;

exports[`<InlineAlert /> with details property renders a banner alert, with no intent, with details 1`] = `
.emotion-4 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
}

.emotion-2 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-0 {
  font-family: Motiva Sans;
  color: #5B5D61;
  font-size: 0.875rem;
  line-height: 1.5;
}

<div
  className="emotion-4 emotion-5"
  display="grid"
>
  <div
    className="emotion-2 emotion-3"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
    <span
      className="emotion-0 emotion-1"
      color="monochrome.grey80"
      fontSize="size4"
    >
       
      Here are some alert details
    </span>
  </div>
</div>
`;

exports[`<InlineAlert /> with details property renders a banner alert, with success intent, with details 1`] = `
.emotion-6 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #5B5D61;
  font-size: 0.875rem;
  line-height: 1.5;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-8 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr;
}

.emotion-0 {
  color: #29863A;
  font-size: 0.875rem;
}

<div
  className="emotion-8 emotion-9"
  display="grid"
>
  <div
    aria-label="success"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-tick-circle emotion-0 emotion-1"
      color="intent.success.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-6 emotion-7"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
    <span
      className="emotion-4 emotion-5"
      color="monochrome.grey80"
      fontSize="size4"
    >
       
      Here are some alert details
    </span>
  </div>
</div>
`;

exports[`<InlineAlert /> with details property renders a banner alert, with warning intent, with details 1`] = `
.emotion-6 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #5B5D61;
  font-size: 0.875rem;
  line-height: 1.5;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-8 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr;
}

.emotion-0 {
  color: #C55205;
  font-size: 0.875rem;
}

<div
  className="emotion-8 emotion-9"
  display="grid"
>
  <div
    aria-label="warning"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-warning-sign emotion-0 emotion-1"
      color="intent.warning.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-6 emotion-7"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
    <span
      className="emotion-4 emotion-5"
      color="monochrome.grey80"
      fontSize="size4"
    >
       
      Here are some alert details
    </span>
  </div>
</div>
`;

exports[`<InlineAlert /> with onClose property renders a banner alert, with danger intent, with an on close 1`] = `
.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-10 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  position: relative;
  text-align: center;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  font-size: 0.75rem;
  height: 1.5rem;
  padding: 0 0.75rem;
  background-color: transparent;
  border: 1px solid transparent;
  color: #081D17;
  font-weight: 500;
  line-height: 1;
  border: 1px solid transparent;
  border-radius: 3px;
}

.emotion-10:disabled {
  background: #EBECEC;
  border: 1px solid transparent;
  color: rgba(120,121,125,0.7);
  cursor: not-allowed;
}

.emotion-10:enabled {
  cursor: pointer;
}

.emotion-10 i {
  vertical-align: middle;
}

.emotion-10 i {
  width: 1.5rem;
  font-size: 0.6875rem;
}

.emotion-10 i:first-child {
  margin-left: -0.75rem;
}

.emotion-10 i:last-child {
  margin-right: -0.75rem;
}

.emotion-10:disabled {
  background-color: transparent;
  color: rgba(120,121,125,0.7);
}

.emotion-10:enabled:active {
  background-color: #F8F9F9;
  color: #134537;
}

.emotion-10:enabled:focus {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #000;
  color: #000;
}

.emotion-10:enabled:hover {
  color: #134537;
}

.emotion-10::-moz-focus-inner {
  border: 0;
}

.emotion-10:focus {
  border: 1px solid transparent;
}

.emotion-10:focus:hover {
  border: 1px solid transparent;
}

.emotion-10 i {
  color: 'inherit';
}

.emotion-8 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 0fr;
  visibility: visible;
}

.emotion-6 {
  font-size: larger;
}

.emotion-14 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr 0fr;
}

.emotion-0 {
  color: #D93838;
  font-size: 0.875rem;
}

<div
  className="emotion-14 emotion-15"
  display="grid"
  onClose={[Function]}
>
  <div
    aria-label="danger"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-error emotion-0 emotion-1"
      color="intent.danger.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-4 emotion-5"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <button
      aria-label="close"
      className="button--minimal emotion-10 emotion-11"
      disabled={false}
      fontWeight="medium"
      onClick={[Function]}
      size="small"
    >
      <div
        className="emotion-8 emotion-9"
        display="inline-grid"
      >
        <i
          className="ar ar-cross emotion-6 emotion-1"
          fontSize="larger"
        />
      </div>
    </button>
  </div>
</div>
`;

exports[`<InlineAlert /> with onClose property renders a banner alert, with info intent, with an on close 1`] = `
.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-10 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  position: relative;
  text-align: center;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  font-size: 0.75rem;
  height: 1.5rem;
  padding: 0 0.75rem;
  background-color: transparent;
  border: 1px solid transparent;
  color: #081D17;
  font-weight: 500;
  line-height: 1;
  border: 1px solid transparent;
  border-radius: 3px;
}

.emotion-10:disabled {
  background: #EBECEC;
  border: 1px solid transparent;
  color: rgba(120,121,125,0.7);
  cursor: not-allowed;
}

.emotion-10:enabled {
  cursor: pointer;
}

.emotion-10 i {
  vertical-align: middle;
}

.emotion-10 i {
  width: 1.5rem;
  font-size: 0.6875rem;
}

.emotion-10 i:first-child {
  margin-left: -0.75rem;
}

.emotion-10 i:last-child {
  margin-right: -0.75rem;
}

.emotion-10:disabled {
  background-color: transparent;
  color: rgba(120,121,125,0.7);
}

.emotion-10:enabled:active {
  background-color: #F8F9F9;
  color: #134537;
}

.emotion-10:enabled:focus {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #000;
  color: #000;
}

.emotion-10:enabled:hover {
  color: #134537;
}

.emotion-10::-moz-focus-inner {
  border: 0;
}

.emotion-10:focus {
  border: 1px solid transparent;
}

.emotion-10:focus:hover {
  border: 1px solid transparent;
}

.emotion-10 i {
  color: 'inherit';
}

.emotion-8 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 0fr;
  visibility: visible;
}

.emotion-6 {
  font-size: larger;
}

.emotion-0 {
  color: #4263EB;
  font-size: 0.875rem;
}

.emotion-14 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr 0fr;
}

<div
  className="emotion-14 emotion-15"
  display="grid"
  onClose={[Function]}
>
  <div
    aria-label="info"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-info-sign emotion-0 emotion-1"
      color="intent.info.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-4 emotion-5"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <button
      aria-label="close"
      className="button--minimal emotion-10 emotion-11"
      disabled={false}
      fontWeight="medium"
      onClick={[Function]}
      size="small"
    >
      <div
        className="emotion-8 emotion-9"
        display="inline-grid"
      >
        <i
          className="ar ar-cross emotion-6 emotion-1"
          fontSize="larger"
        />
      </div>
    </button>
  </div>
</div>
`;

exports[`<InlineAlert /> with onClose property renders a banner alert, with no intent, with an on close 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-10 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 1fr 0fr;
}

.emotion-8 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-6 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  position: relative;
  text-align: center;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  font-size: 0.75rem;
  height: 1.5rem;
  padding: 0 0.75rem;
  background-color: transparent;
  border: 1px solid transparent;
  color: #081D17;
  font-weight: 500;
  line-height: 1;
  border: 1px solid transparent;
  border-radius: 3px;
}

.emotion-6:disabled {
  background: #EBECEC;
  border: 1px solid transparent;
  color: rgba(120,121,125,0.7);
  cursor: not-allowed;
}

.emotion-6:enabled {
  cursor: pointer;
}

.emotion-6 i {
  vertical-align: middle;
}

.emotion-6 i {
  width: 1.5rem;
  font-size: 0.6875rem;
}

.emotion-6 i:first-child {
  margin-left: -0.75rem;
}

.emotion-6 i:last-child {
  margin-right: -0.75rem;
}

.emotion-6:disabled {
  background-color: transparent;
  color: rgba(120,121,125,0.7);
}

.emotion-6:enabled:active {
  background-color: #F8F9F9;
  color: #134537;
}

.emotion-6:enabled:focus {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #000;
  color: #000;
}

.emotion-6:enabled:hover {
  color: #134537;
}

.emotion-6::-moz-focus-inner {
  border: 0;
}

.emotion-6:focus {
  border: 1px solid transparent;
}

.emotion-6:focus:hover {
  border: 1px solid transparent;
}

.emotion-6 i {
  color: 'inherit';
}

.emotion-4 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 0fr;
  visibility: visible;
}

.emotion-2 {
  font-size: larger;
}

<div
  className="emotion-10 emotion-11"
  display="grid"
  onClose={[Function]}
>
  <div
    className="emotion-0 emotion-1"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
  <div
    className="emotion-8 emotion-9"
  >
    <button
      aria-label="close"
      className="button--minimal emotion-6 emotion-7"
      disabled={false}
      fontWeight="medium"
      onClick={[Function]}
      size="small"
    >
      <div
        className="emotion-4 emotion-5"
        display="inline-grid"
      >
        <i
          className="ar ar-cross emotion-2 emotion-3"
          fontSize="larger"
        />
      </div>
    </button>
  </div>
</div>
`;

exports[`<InlineAlert /> with onClose property renders a banner alert, with success intent, with an on close 1`] = `
.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-10 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  position: relative;
  text-align: center;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  font-size: 0.75rem;
  height: 1.5rem;
  padding: 0 0.75rem;
  background-color: transparent;
  border: 1px solid transparent;
  color: #081D17;
  font-weight: 500;
  line-height: 1;
  border: 1px solid transparent;
  border-radius: 3px;
}

.emotion-10:disabled {
  background: #EBECEC;
  border: 1px solid transparent;
  color: rgba(120,121,125,0.7);
  cursor: not-allowed;
}

.emotion-10:enabled {
  cursor: pointer;
}

.emotion-10 i {
  vertical-align: middle;
}

.emotion-10 i {
  width: 1.5rem;
  font-size: 0.6875rem;
}

.emotion-10 i:first-child {
  margin-left: -0.75rem;
}

.emotion-10 i:last-child {
  margin-right: -0.75rem;
}

.emotion-10:disabled {
  background-color: transparent;
  color: rgba(120,121,125,0.7);
}

.emotion-10:enabled:active {
  background-color: #F8F9F9;
  color: #134537;
}

.emotion-10:enabled:focus {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #000;
  color: #000;
}

.emotion-10:enabled:hover {
  color: #134537;
}

.emotion-10::-moz-focus-inner {
  border: 0;
}

.emotion-10:focus {
  border: 1px solid transparent;
}

.emotion-10:focus:hover {
  border: 1px solid transparent;
}

.emotion-10 i {
  color: 'inherit';
}

.emotion-8 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 0fr;
  visibility: visible;
}

.emotion-6 {
  font-size: larger;
}

.emotion-14 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr 0fr;
}

.emotion-0 {
  color: #29863A;
  font-size: 0.875rem;
}

<div
  className="emotion-14 emotion-15"
  display="grid"
  onClose={[Function]}
>
  <div
    aria-label="success"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-tick-circle emotion-0 emotion-1"
      color="intent.success.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-4 emotion-5"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <button
      aria-label="close"
      className="button--minimal emotion-10 emotion-11"
      disabled={false}
      fontWeight="medium"
      onClick={[Function]}
      size="small"
    >
      <div
        className="emotion-8 emotion-9"
        display="inline-grid"
      >
        <i
          className="ar ar-cross emotion-6 emotion-1"
          fontSize="larger"
        />
      </div>
    </button>
  </div>
</div>
`;

exports[`<InlineAlert /> with onClose property renders a banner alert, with warning intent, with an on close 1`] = `
.emotion-4 {
  font-family: Motiva Sans;
  color: #0F0F10;
  font-size: 0.875rem;
  font-weight: 500;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-10 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  position: relative;
  text-align: center;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  font-size: 0.75rem;
  height: 1.5rem;
  padding: 0 0.75rem;
  background-color: transparent;
  border: 1px solid transparent;
  color: #081D17;
  font-weight: 500;
  line-height: 1;
  border: 1px solid transparent;
  border-radius: 3px;
}

.emotion-10:disabled {
  background: #EBECEC;
  border: 1px solid transparent;
  color: rgba(120,121,125,0.7);
  cursor: not-allowed;
}

.emotion-10:enabled {
  cursor: pointer;
}

.emotion-10 i {
  vertical-align: middle;
}

.emotion-10 i {
  width: 1.5rem;
  font-size: 0.6875rem;
}

.emotion-10 i:first-child {
  margin-left: -0.75rem;
}

.emotion-10 i:last-child {
  margin-right: -0.75rem;
}

.emotion-10:disabled {
  background-color: transparent;
  color: rgba(120,121,125,0.7);
}

.emotion-10:enabled:active {
  background-color: #F8F9F9;
  color: #134537;
}

.emotion-10:enabled:focus {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #000;
  color: #000;
}

.emotion-10:enabled:hover {
  color: #134537;
}

.emotion-10::-moz-focus-inner {
  border: 0;
}

.emotion-10:focus {
  border: 1px solid transparent;
}

.emotion-10:focus:hover {
  border: 1px solid transparent;
}

.emotion-10 i {
  color: 'inherit';
}

.emotion-8 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 0fr;
  visibility: visible;
}

.emotion-6 {
  font-size: larger;
}

.emotion-14 {
  font-family: Motiva Sans;
  padding: 0;
  box-sizing: border-box;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 1rem;
  grid-row-gap: 0.5rem;
  grid-template-columns: 0fr 1fr 0fr;
}

.emotion-0 {
  color: #C55205;
  font-size: 0.875rem;
}

<div
  className="emotion-14 emotion-15"
  display="grid"
  onClose={[Function]}
>
  <div
    aria-label="warning"
    className="emotion-2 emotion-3"
  >
    <i
      className="ar ar-warning-sign emotion-0 emotion-1"
      color="intent.warning.default"
      fontSize="size4"
    />
  </div>
  <div
    className="emotion-4 emotion-5"
    color="monochrome.grey100"
    fontSize="size4"
    fontWeight="medium"
  >
    Hey look, an alert!
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <button
      aria-label="close"
      className="button--minimal emotion-10 emotion-11"
      disabled={false}
      fontWeight="medium"
      onClick={[Function]}
      size="small"
    >
      <div
        className="emotion-8 emotion-9"
        display="inline-grid"
      >
        <i
          className="ar ar-cross emotion-6 emotion-1"
          fontSize="larger"
        />
      </div>
    </button>
  </div>
</div>
`;
